rest/networkProtocols/TheThingsNetwork.js

```diff
@@ -17,25 +17,37 @@ module.exports = {
       oauthUrl: 'https://account.thethingsnetwork.org/users/authorize',
       protocolHandlerNetworkFields: [
         {
-          fieldName: 'clientId',
-          fieldDesc: 'Client Id',
-          fieldHelp: 'The client id chosen when registering LPWan',
-          fieldType: 'string',
-          displayWithQueryParameter: ''
+          name: 'clientId',
+          description: 'Client Id',
+          help: 'The client id chosen when registering LPWan',
+          type: 'string',
+          label: 'Client ID',
+          value: '',
+          required: true,
+          placeholder: 'TTN Client ID',
+          oauthQueryParameter: ''
         },
         {
-          fieldName: 'clientSecret',
-          fieldDesc: 'Client Secret',
-          fieldHelp: 'The client secret provided when registering LPWan',
-          fieldType: 'string',
-          displayWithQueryParameter: ''
+          name: 'clientSecret',
+          description: 'Client Secret',
+          help: 'The client secret provided when registering LPWan',
+          type: 'string',
+          label: 'Client Secret',
+          value: '',
+          required: true,
+          placeholder: 'TTN Client Secret',
+          oauthQueryParameter: ''
         },
         {
-          fieldName: 'accessCode',
```

Comment:
I personally think including code as one of the fields will be confusing to the user (even if read only). They will have no idea what it is, or how to use it. I know that @JimCAtSentosa likes it for trouble shooting purposes, but I would propose that we display code in the error message to the user, for trouble shooting purposes when there is a problem

What you you think @dschrimpsher

Reply:
I just provide it.  I leave it up to you "UI people" to figure out what to show.  :-)